File: robocode-tankroyale-server/src/main/java/dev/robocode/tankroyale/server/events/BulletHitWallEvent.java
package dev.robocode.tankroyale.server.events;

import lombok.Value;
import dev.robocode.tankroyale.server.model.Bullet;

/**
 * Event sent when a bullet has hit the wall of the battle arena.
 * 
 * @author Flemming N. Larsen
 */
@Value
public class BulletHitWallEvent implements Event {

	/** Turn number when event occurred */
	int turnNumber;

	/** Bullet that missed */
	Bullet bullet;
}

File: server/src/main/kotlin/dev/robocode/tankroyale/server/event/BulletHitWallEvent.kt
package dev.robocode.tankroyale.server.event

import dev.robocode.tankroyale.server.model.IBullet

/** Event sent when a bullet has hit the wall of the battle arena. */
class BulletHitWallEvent(
    /** Turn number when event occurred */
    override val turnNumber: Int,

    /** Bullet that missed */
    val bullet: IBullet
) : Event()